server/src/main/java/io/crate/planner/optimizer/collect/Rule.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to Crate under one or more contributor license agreements.
+ * See the NOTICE file distributed with this work for additional
+ * information regarding copyright ownership.  Crate licenses this file
+ * to you under the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied.  See the License for the specific language governing
+ * permissions and limitations under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial
+ * agreement.
+ */
+
+package io.crate.planner.optimizer.collect;
+
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.Functions;
+import io.crate.planner.optimizer.matcher.Captures;
+import io.crate.planner.optimizer.matcher.Pattern;
+import org.elasticsearch.Version;
+
+public interface Rule<T> {
+
+    Pattern<T> pattern();
+
+    boolean isEnabled();
+
+    void setEnabled(boolean enabled);
```

Comment:
Looks like this is not used yet. Should we remove this + `isEnabled` altogether for now?

Reply:
Right. Maybe we should group all cast rules and expose one session setting to enable/disable all?

Reply:
> Right. Maybe we should group all cast rules and expose one session setting to enable/disable all?

Not sure. I'd leave it out and postpone the decision how to handle this for later.